buildlib/test.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash -eExl
+#
+# Testing script for SparkUCX
+#
+# Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+
+# Required parameter space separated list of nodes
+NODELIST=${NODELIST:=echo "NODELIST is required"; exit 1}
+
+# Optional parameters
+
+# Path to UCX installation
+UCX_LIB=${UCX_LIB:=$PWD/ucx/build/lib}
+
+# Path to SparkUCX jar to test
+SPARK_UCX_JAR=${SPARK_UCX_JAR:=$PWD/spark-ucx-1.0-for-spark-2.4.0-jar-with-dependencies.jar}
+
+# Number of processes per instance
+PROCESSES_PER_INSTANCE=${PROCESSES_PER_INSTANCE:=1}
+
+# Number number of executor within the test machine
+EXECUTOR_NUMBER=${EXECUTOR_NUMBER:=$((1 + RANDOM % 10))}
+
+# Unique job id
+JOB_ID=${SLURM_JOB_ID:="sparkucx-test-$(cat /proc/sys/kernel/random/uuid)"}
+
+# Separate directory for this test where configs will be added. For now it should be on NFS
+SCRATCH_DIRECTORY=${SCRATCH_DIRECTORY:=$PWD/${JOB_ID}}
+
+# Directory to use for "scratch" space in Spark, including map output files and RDDs that get stored on disk.
+# This should be on a fast, local disk in your system.
+# It can also be a comma-separated list of multiple directories on different disks.
+SPARK_LOCAL_DIRS=${SPARK_LOCAL_DIRS:="/scrap/sparkucxtest/sparkucx-${JOB_ID}"}
+
+# Number of cores per spark process
+SPARK_WORKER_CORES=${SPARK_WORKER_CORES:=10}
```

Comment:
can we have 1 by default?

Reply:
Let's make at least 2. Many errors occurred before when there were multiple threads.

Reply:
ok
can we also set CPU affinity to EXECUTOR_NUMER*2,EXECUTOR_NUMER*2+1,